cmd/limactl/delete.go

```diff
@@ -53,9 +55,16 @@ func deleteInstance(inst *store.Instance, force bool) error {
 
 	stopInstanceForcibly(inst)
 
+	if inst.VMType == limayaml.WSL {
+		if _, err := executil.RunUTF16leCommand([]string{"wsl", "--unregister", inst.DistroName}); err != nil {
```

Comment:
This could be a good place to use the Unregister interface, that was introduced with Virtualbox.

* https://github.com/lima-vm/lima/pull/1277

Maybe we should add that to Lima first (as a no-op), independent of any driver that might use it ?

```diff
--- a/pkg/driver/driver.go
+++ b/pkg/driver/driver.go
@@ -35,6 +35,12 @@ type Driver interface {
 	// It returns error if there are any errors during Stop
 	Stop(_ context.Context) error
 
+	// Register adds the instance to the registry.
+	Register(_ context.Context) error
+
+	// Unregister removes the instance from the registry.
+	Unregister(_ context.Context) error
+
 	ChangeDisplayPassword(_ context.Context, password string) error
 
 	GetDisplayConnection(_ context.Context) (string, error)

```

Reply:
Good idea. I'll port your Unregister changes to a separate PR and comment back when it's ready 👍